packages/core/admin/admin/src/content-manager/components/BlocksEditor/hooks/useBlocksStore.js

```diff
@@ -255,18 +259,33 @@ export function useBlocksStore() {
          */
         const [createdNode] = Editor.parent(editor, editor.selection.anchor.path);
         Transforms.removeNodes(editor, editor.selection);
-        Transforms.insertNodes(editor, {
-          type: 'paragraph',
-          children: createdNode.children,
-        });
+
+        /**
+         * Check if after the current position there is another node
+         */
+        const hasNextNode = editor.children.length - anchorPathInitialPosition[0] > 1;
```

Comment:
Instead if manually finding next node, there is a `editor.next` method available, may be we can use that - https://docs.slatejs.org/api/nodes/editor#editor.next-less-than-t-extends-descendant-greater-than-editor-editor-options-greater-than-nodeentry

Reply:
I tried different options yesterday and it takes all the day and the only way it worked is the one I found, I know it is not super elegant but it fixes the issue for the moment, if you want I can add a todo to remember we can find a better way

Reply:
thanks @madhurisandbhor for the suggestion

Reply:
Sorry @madhurisandbhor but I need to revert back my code because 
`const hasNextNode = Boolean(Editor.next(editor, { at: editor.selection }))`;
doesn't work if the distance between the selection and the last line in the editor is greater than 1